src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.AnyUnix.cs

```diff
@@ -0,0 +1,1793 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Security;
+
+namespace System
+{
+    public sealed partial class TimeZoneInfo
+    {
+        private const string DefaultTimeZoneDirectory = "/usr/share/zoneinfo/";
+        private const string TimeZoneEnvironmentVariable = "TZ";
+
+        // UTC aliases per https://github.com/unicode-org/cldr/blob/master/common/bcp47/timezone.xml
+        // Hard-coded because we need to treat all aliases of UTC the same even when globalization data is not available.
+        // (This list is not likely to change.)
+        private static readonly string[] s_UtcAliases = new[] {
+            "Etc/UTC",
+            "Etc/UCT",
+            "Etc/Universal",
+            "Etc/Zulu",
+            "UCT",
+            "UTC",
+            "Universal",
+            "Zulu"
+        };
+
+        private static readonly TimeZoneInfo s_utcTimeZone = CreateUtcTimeZone();
+
+        private TimeZoneInfo(byte[] data, string id, bool dstDisabled)
```

Comment:
I am fairly sure this is specific to .Unix and should be moved there.

Reply:
This function is called in `GetTimeZoneFromTzData` which is responsible for returning a new TimeZone when given the time zone byte data as well as the time zone ID, and that is useful for `TimeZoneInfo.Android.cs` as well. I don't think Android will need a unique implementation as long as the proper byte data is passed, and that is handled by `AndroidTzData.GetTimeZoneData`.